src/main/scala/com/fulcrumgenomics/bam/ZipperBams.scala

```diff
@@ -0,0 +1,191 @@
+      if (tagInfo.hasRevsOrRevcomps && ms.exists(_.negativeStrand)) {
+        u.asSam.reverseComplement(tagInfo.setToRevcomp, tagInfo.setToReverse, true)
+      }
+
+      // And deal with the negative strand mappings
+      ms.iterator.filter(_.negativeStrand).foreach(m => copyTags(u, m))
+    }
+
+    mapped
+  }
+
+  /** Copies all tags from read to another. */
+  def copyTags(src: SamRecord, dest: SamRecord): Unit = {
+    src.attributes.foreach { case (tag, value) => dest(tag) = value }
+  }
+}
+
+@clp(group=ClpGroups.SamOrBam, description=
+  """
+    |Zips together an unmapped and mapped BAM to transfer metadata into the output BAM.
+    |
+    |Both the unmapped and mapped BAMs _must_ be a) queryname sorted or grouped (i.e. all records with the same
+    |name are grouped together in the file), and b) have the same ordering of querynames.  If either of these are
+    |violated the output is undefined!
+    |
+    |By default the mapped BAM is read from stdin and the output BAM is written to stdout. This can be changed
+    |using the `--input/-i` and `--output/-o` options.
+    |
+    |By default the output BAM file is emitted in the same order as the input BAMs.  This can be overridden
+    |using the `--sort` option, though in practice it may be faster to do the following:
+    |
+    |  `fgbio --compression 0 ZipperBams -i mapped.bam -u unmapped.bam -r ref.fa | samtools sort -@ 16`
+    |
+    |
+  """)
+class ZipperBams
+( @arg(flag='i', doc="Mapped SAM or BAM.") val input: PathToBam = Io.StdIn,
+  @arg(flag='u', doc="Unmapped SAM or BAM.") val unmapped: PathToBam,
+  @arg(flag='r', doc="Path to the reference used in alignment.") val ref: PathToFasta,
+  @arg(flag='o', doc="Output SAM or BAM file.") val output: PathToBam = Io.StdOut,
+  @arg(doc="Tags to remove from the mapped BAM records.", minElements=0) val tagsToRemove: IndexedSeq[String] = IndexedSeq.empty,
+  @arg(doc="Set of optional tags to reverse on reads mapped to the negative strand.", minElements=0) val tagsToReverse: IndexedSeq[String] = IndexedSeq.empty,
+  @arg(doc="Set of optional tags to reverse complement on reads mapped to the negative strand.", minElements=0) val tagsToRevcomp: IndexedSeq[String] = IndexedSeq.empty,
+  @arg(flag='s', doc="Sort the output BAM into the given order.") val sort: Option[SamOrder] = None,
+  @arg(flag='b', doc="Buffer this many read-pairs while reading the input BAMs.") val buffer: Int = 5000
+) extends FgBioTool {
+  import ZipperBams._
+
+  override def execute(): Unit = {
+    val unmappedSource = SamSource(unmapped)
+    val mappedSource   = SamSource(input)
+    checkSort(unmappedSource, unmapped, "unmapped")
+    checkSort(mappedSource,  input, "input")
+
+    val dict    = SAMSequenceDictionaryExtractor.extractDictionary(this.ref)
+    val header  = buildOutputHeader(unmappedSource.header, mappedSource.header, this.sort, dict)
+    val out     = SamWriter(output, header, sort=this.sort, async=true)
```

Comment:
Why explicitly set `async=true` here? I am seeing a trend of having this always on, so why not just set it as always on at the tool level?

Reply:
Should probably not do this .. there's no easy way for a single tool to override the default though as it's off in another class that the tool doesn't see directly.